File: leetcode/1557. Minimum Number of Vertices to Reach All Nodes.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<bool> has_parent(n, false);
        for(auto edge: edges){
            has_parent[edge[1]] = true;
        }
        vector<int> ans;
        for (int i = 0; i < n; i++){
            if(!has_parent[i]){
                ans.push_back(i);
            }
        }
        return ans;
    }
};

File: leetcode/1557. Minimum Number of Vertices to Reach All Nodes.py
from typing import List
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        has_parent = [False] * n
        for edge in edges:
            has_parent[edge[1]] = True
        ans = []
        for i in range(n):
            if has_parent[i] is False:
                ans.append(i)
        return ans